app/models/middleware_server.rb

```diff
@@ -33,6 +37,21 @@ def evaluate_alert(alert_id, event)
   end
 
   def in_domain?
-    !middleware_server_group.nil?
+    middleware_server_group
```

Comment:
I'm finding it hard to follow (maybe it's just me :)) - how does this return boolean?